test/new_tests/test_query_apply.py

```diff
@@ -508,3 +509,57 @@ def _no_items_have_been_applied(self):
             assert bins["name"] != "aerospike"
         else:
             assert bins["name"] == "no_set_name"
+
+    def test_fail_if_operations_exist_before_apply(self):
```

Comment:
Shouldn't these tests be checking query.foreach() and query.results()?

Reply:
I included the wrong test cases in this branch, I removed them and added the correct ones, so these comments should not be relevant anymore.